File: ds-worker/src/main/java/com/example/dsworker/entity/DBSchema.java
package com.example.dsworker.entity;

import lombok.Data;

import java.util.List;

@Data
public class DBSchema {

    @Data
    static public class DBTable {
        private String name;
        private String remark;
        private List<DBColumn> columns;
        private List<String> primaryKeys;
    }

    @Data
    static public class DBColumn {
        private String name;
        private String typ;
        private int nullable;
        private String remark;
    }

    private String dbName;
    private String dbProductName;
    private String dbProductVersion;
    private String dbDriverName;
    private List<DBTable> tables;

}

File: chat-analysis/src/main/kotlin/org/inet/aet/chatanalysis/entity/DBSchema.kt
package org.inet.aet.chatanalysis.entity


data class DBTable (
    var name: String,
    var remark: String,
    val columns: List<DBColumn>,
    val primaryKeys: List<String>
)

data class DBColumn (
    var name: String,
    var typ: String,
    var nullable: Int,
    var remark: String,
)

data class DBSchema (
    var dbName: String,
    var dbProductName: String,
    var dbProductVersion: String,
    var dbDriverName: String,
    val tables: List<DBTable>,
)